server/routes/bookedtoday/arrivals/searchforexisting/index.ts

```diff
@@ -14,50 +12,40 @@ import redirectIfDisabledMiddleware from '../../../../middleware/redirectIfDisab
 import { State } from '../state'
 import searchRoutes from './search'
+import Routes from '../../../../utils/routeBuilder'
 
 export default function routes(services: Services): Router {
-  const router = express.Router()
-
   const checkSearchDetailsPresent = State.searchDetails.ensurePresent('/')
   const checkNewArrivalPresent = State.newArrival.ensurePresent('/')
 
-  const get = (path: string, handlers: RequestHandler[]) =>
-    router.get(
-      `/prisoners/:id/search-for-existing-record${path}`,
-      authorisationForUrlMiddleware([Role.PRISON_RECEPTION]),
-      [
-        redirectIfDisabledMiddleware(config.confirmNoIdentifiersEnabled),
-        ...handlers.map(handler => asyncMiddleware(handler)),
-      ]
-    )
-
-  const post = (path: string, handlers: RequestHandler[]) =>
-    router.post(
-      `/prisoners/:id/search-for-existing-record${path}`,
-      authorisationForUrlMiddleware([Role.PRISON_RECEPTION]),
-      [
-        redirectIfDisabledMiddleware(config.confirmNoIdentifiersEnabled),
-        ...handlers.map(handler => asyncMiddleware(handler)),
-      ]
-    )
-
-  const multipleExistingRecordsFoundController = new MultipleExistingRecordsFoundController(
-    services.expectedArrivalsService
-  )
-  get('/possible-records-found', [checkSearchDetailsPresent, multipleExistingRecordsFoundController.view()])
-  post('/possible-records-found', [
-    checkSearchDetailsPresent,
-    validationMiddleware(MatchedRecordSelectionValidation),
-    multipleExistingRecordsFoundController.submit(),
-  ])
+  const basePath = `/prisoners/:id/search-for-existing-record`
 
-  const singleExistingRecordFoundController = new SingleExistingRecordFoundController()
-  get('/record-found', [checkNewArrivalPresent, checkSearchDetailsPresent, singleExistingRecordFoundController.view()])
+  const multipleMatchFoundController = new MultipleExistingRecordsFoundController(services.expectedArrivalsService)
+  const singleMatchFoundController = new SingleExistingRecordFoundController()
+  const noMatchFoundController = new NoExistingRecordsFoundController()
 
-  const noExistingRecordsFoundController = new NoExistingRecordsFoundController()
-  get('/no-record-found', [noExistingRecordsFoundController.view()])
-
-  router.use(searchRoutes(services))
-
-  return router
+  return Routes.forRole(Role.PRISON_RECEPTION)
+    .get(
+      `${basePath}/possible-records-found`,
+      redirectIfDisabledMiddleware(config.confirmNoIdentifiersEnabled),
+      checkSearchDetailsPresent,
+      multipleMatchFoundController.view()
+    )
+    .post(
+      `${basePath}/possible-records-found`,
+      redirectIfDisabledMiddleware(config.confirmNoIdentifiersEnabled),
+      checkSearchDetailsPresent,
+      validationMiddleware(MatchedRecordSelectionValidation),
+      multipleMatchFoundController.submit()
+    )
+    .get(
+      `${basePath}/record-found`,
+      redirectIfDisabledMiddleware(config.confirmNoIdentifiersEnabled),
```

Comment:
think we can remove this feature toggle now as the functionality is all present, just need to check normal feature toggle is in place for all routes